Subject: [PATCH] soong: add TARGET_SPECIFIC_HEADER_PATH to config

diff --git a/android/config.go b/android/config.go
index aa0ac82..00833c2 100644
--- a/android/config.go
+++ b/android/config.go
@@ -859,3 +859,7 @@
 		return nil
 	}
 }
+
+func (c *deviceConfig) TargetHeaderPathDir() string {
+	return String(c.config.productVariables.TargetHeaderPath)
+}
diff --git a/android/variable.go b/android/variable.go
index 6113063..2165ff6 100644
--- a/android/variable.go
+++ b/android/variable.go
@@ -225,6 +225,8 @@

	Override_rs_driver *string `json:",omitempty"`

+	TargetHeaderPath *string `json:",omitempty"`
+
 	DeviceKernelHeaders []string `json:",omitempty"`
 	DistDir             *string  `json:",omitempty"`
 
diff --git a/cc/library.go b/cc/library.go
index 4a173a5..0744dfa 100644
--- a/cc/library.go
+++ b/cc/library.go
@@ -319,6 +319,9 @@
 }
 
 func (library *libraryDecorator) compilerFlags(ctx ModuleContext, flags Flags, deps PathDeps) Flags {
+	if len(ctx.DeviceConfig().TargetHeaderPathDir()) > 0 {
+		flags.GlobalFlags = append(flags.GlobalFlags, "-I" + ctx.DeviceConfig().TargetHeaderPathDir())
+	}
 	exportIncludeDirs := library.flagExporter.exportedIncludes(ctx)
 	if len(exportIncludeDirs) > 0 {
 		f := includeDirsToFlags(exportIncludeDirs)
