Subject: [PATCH] soong: Default kati jobs to NumCPU()

diff --git a/cmd/multiproduct_kati/main.go b/cmd/multiproduct_kati/main.go
index 06c56264..3bedfa18 100644
--- a/cmd/multiproduct_kati/main.go
+++ b/cmd/multiproduct_kati/main.go
@@ -33,15 +33,9 @@ import (
 	"android/soong/zip"
 )
 
-// We default to number of cpus / 4, which seems to be the sweet spot for my
-// system. I suspect this is mostly due to memory or disk bandwidth though, and
-// may depend on the size ofthe source tree, so this probably isn't a great
-// default.
+// We default to number of cpus.
 func detectNumJobs() int {
-	if runtime.NumCPU() < 4 {
-		return 1
-	}
-	return runtime.NumCPU() / 4
+	return (runtime.NumCPU() - 1)
 }
 
 var numJobs = flag.Int("j", detectNumJobs(), "number of parallel kati jobs")
