Subject: [PATCH] build: add TARGET_SPECIFIC_HEADER_PATH to config

diff --git a/core/soong_config.mk b/core/soong_config.mk
index 1b012ff..b01fe99 100644
--- a/core/soong_config.mk
+++ b/core/soong_config.mk
@@ -123,6 +123,7 @@
 $(call add_json_list, Platform_systemsdk_versions,       $(PLATFORM_SYSTEMSDK_VERSIONS))
 $(call add_json_bool, Malloc_not_svelte,                 $(call invert_bool,$(filter true,$(MALLOC_SVELTE))))
 $(call add_json_str,  Override_rs_driver,                $(OVERRIDE_RS_DRIVER))
+$(call add_json_str,  TargetHeaderPath,                  $(TARGET_SPECIFIC_HEADER_PATH))
 
 $(call add_json_bool, Treble_linker_namespaces,          $(filter true,$(PRODUCT_TREBLE_LINKER_NAMESPACES)))
 $(call add_json_bool, Enforce_vintf_manifest,            $(filter true,$(PRODUCT_ENFORCE_VINTF_MANIFEST)))
