Subject: [PATCH] Fix Jack

diff --git a/core/config.mk b/core/config.mk
index 1ecf146..0ac39bd 100644
--- a/core/config.mk
+++ b/core/config.mk
@@ -764,7 +764,7 @@ APICHECK_CLASSPATH_ENTRIES := \
     )
 APICHECK_CLASSPATH := $(subst $(space),:,$(strip $(APICHECK_CLASSPATH_ENTRIES)))
 
-APICHECK_COMMAND := $(APICHECK) -JXmx1024m -J"classpath $(APICHECK_CLASSPATH)"
+APICHECK_COMMAND := $(APICHECK) -JXmx6144m -J"classpath $(APICHECK_CLASSPATH)"
 
 # Boolean variable determining if the whitelist for compatible properties is enabled
 PRODUCT_COMPATIBLE_PROPERTY := false
