Subject: [PATCH] vendor: fix ro.adb.secure for vendor-building devices

diff --git a/config/common.mk b/config/common.mk
index 8d17bb1b..f2887d63 100644
--- a/config/common.mk
+++ b/config/common.mk
@@ -31,7 +31,7 @@ endif
 PRODUCT_SYSTEM_DEFAULT_PROPERTIES += ro.adb.secure=0
 else
 # Enable ADB authentication
-PRODUCT_SYSTEM_DEFAULT_PROPERTIES += ro.adb.secure=1
+PRODUCT_SYSTEM_DEFAULT_PROPERTIES += ro.adb.secure=0
 endif
 
 # Backup Tool
